### PR TITLE
feat: add symbols to orama

### PR DESCRIPTION
**Pull request by crowlKats**
- [x] add deploying of the index

search itself will be added next week

### CODE REVIEW

File: api/src/analysis.rs [hunk 1]
```diff
@@ -227,17 +228,52 @@ async fn analyze_package_inner(
...
+      browser: None,
+      deno: None,
+      node: None,
+      workerd: None,
+      bun: None,
+    },
+    registry_url.to_string(),
+  );
+  let search_index = deno_doc::html::generate_search_index(&ctx);
+  let doc_search_json = if let serde_json::Value::Object(mut obj) = search_index
```
Comment (lucacasonato) [inline api/src/analysis.rs:265]:
```suggestion
  let doc_search_json = if let serde_json::Value::Object(mut obj) = search_index
```

You can replace this with `search_index.as_object().unwrap()`.

File: api/src/orama.rs [hunk 2]
```diff
@@ -120,6 +126,89 @@ impl OramaClient {
...
+  }
+
+  #[instrument(name = "OramaClient::upsert_symbols", skip(self))]
+  pub fn upsert_symbols(
+    &self,
+    scope_name: &ScopeName,
+    package_name: &PackageName,
+    search: serde_json::Value,
+  ) {
+    let package = format!("{scope_name}/{package_name}");
```
Comment (lucacasonato) [inline api/src/orama.rs:141]: Why two calls, can't we do one?

File: api/src/orama.rs [hunk 3]
```diff
@@ -120,6 +126,89 @@ impl OramaClient {
...
+          } else {
+            unreachable!()
+          }
+        })
+        .collect::<Vec<_>>()
+    } else {
+      unreachable!()
+    };
+
+    let body = serde_json::json!({ "upsert": search });
```
Comment (crowlKats) [inline api/src/orama.rs:191]: TODO: handle too big data size

File: api/src/analysis.rs (hunk 1 above)
Comment (crowlKats) [reply api/src/analysis.rs:265]: no, it returns a reference

File: terraform/scheduler.tf [hunk 4]
```diff
@@ -42,4 +42,12 @@ resource "google_cloud_scheduler_job" "orama_deploy" {
       "Authorization" = "Bearer ${var.orama_package_private_api_key}"
     }
   }
+
+  http_target {
```
Comment (lucacasonato) [inline terraform/scheduler.tf:46]: I don't think you can have two targets. I think you need to create a second `google_cloud_scheduler_job`

Comment (crowlKats) [reply terraform/scheduler.tf:46]: ah damn, yea wasnt sure, was hoping it would work

**Review by lucacasonato — APPROVED** [review]

**Pull request merged**